pgbot/routine.py

```diff
@@ -110,6 +112,102 @@ async def handle_console():
         )
 
 
+async def handle_depression():
+    """
+    Function to handle depression emotion
+    """
+    async with db.DiscordDB("emotions") as db_obj:
+        all_emotions = db_obj.get({})
+
+        try:
+            # Gets the depression emotion dict
+            depression = all_emotions["depression"]
+        except KeyError:
+            depression = {}
+
+        happiness = all_emotions.get("happy", 0)
+        if happiness > 0:
+            # Bot is happy
+            depression["min_of_sadness"] = None
+
+            if depression.get("value", 0) > 0:
+                # If depression is > 0 when bot is happy,
+                # use happiness sigmoid inverse piecewise function
+                try:
+                    depression["min_of_happiness"] = (
+                        emotion.depression_function_curve(
+                            "happy", depression.get("value", 0), inverse=True
+                        )
+                        + num_increment
+                    )
+                except ValueError:
+                    depression["min_of_happiness"] = num_increment
+            else:
+                # If depression is 0 when bot is happy,
+                # start incrementing min of happiness
+                try:
+                    depression["min_of_happiness"] = (
+                        depression.get("min_of_happiness", 0) + num_increment
+                    )
+                except TypeError:
+                    depression["min_of_happiness"] = num_increment
+        else:
+            # Bot is sad
+            depression["min_of_happiness"] = None
+
+            if 0 < depression.get("value", 0) < 100:
+                # If depression in range (0, 100),
+                # Use depression sigmoid inverse piecewise function
+                try:
+                    depression["min_of_sadness"] = (
+                        emotion.depression_function_curve(
+                            "depression", depression.get("value", 0), inverse=True
+                        )
+                        + num_increment
+                    )
+                except ValueError:
+                    depression["min_of_happiness"] += 0.1
+            else:
+                # Otherwise, increment min of sadness
+                try:
+                    depression["min_of_sadness"] = (
+                        depression.get("min_of_sadness", 0) + num_increment
+                    )
+                except TypeError:
+                    depression["min_of_sadness"] = num_increment
+
+        try:
+            # Sets the min value of happiness to num increment
+            if depression.get("min_of_happiness", 0) < 0:
+                depression["min_of_happiness"] = num_increment
+        except TypeError:
```

Comment:
This TypeError check is definately not required

Reply:
depression.get('min_of_happiness', 0) could still return None, and you can't compare `None` to `0`